moonstone/filtering/basics_filtering.py

```diff
@@ -61,3 +63,50 @@ def filter(self) -> pd.DataFrame:
             return self._select_names()
         else:
             return self._exclude_names()
+
+
+class NaNPercentageFiltering(BothAxisFiltering):
+
+    def __init__(self, dataframe: pd.DataFrame, percentage_nan_allowed: Union[int, float] = 80, axis: int = 0):
+        """
+        :param percentage_nan_allowed: maximum percentage of NaN values allowed (between 0 and 100)
+        :param axis: axis to apply filtering (index (0) or columns(1))
+        """
+        self.percentage_nan_allowed = percentage_nan_allowed / 100
+        super().__init__(dataframe, axis=axis)
+
+    def filter(self) -> pd.DataFrame:
```

Comment:
Did you look at the `.dropna()` method? It might simplify the reading as it directly has a `thresh` you can use to determine how many non-na-value you want.

doc: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.dropna.html

Reply:
Oh, right! It didn't come up when I was looking for it during the analysis. But that's great! I'll remove it then!